python/ray/autoscaler/aws/config.py

```diff
@@ -705,15 +725,25 @@ def _client(name, config):
 def _resource(name, config):
     region = config["provider"]["region"]
     aws_credentials = config["provider"].get("aws_credentials", {})
-    return _resource_cache(name, region, **aws_credentials)
-
-
-@lru_cache()
-def _resource_cache(name, region, **kwargs):
-    boto_config = Config(retries={"max_attempts": BOTO_MAX_RETRIES})
-    return boto3.resource(
-        name,
-        region,
-        config=boto_config,
-        **kwargs,
-    )
+    return resource_cache(name, region, **aws_credentials)
+
+
+def _add_cloudwatch_agent_policies(policy_doc, attach_policy_arns):
+    policy_doc["Statement"].extend([{
+        "Action": "sts:AssumeRole",
+        "Effect": "Allow",
+        "Principal": {
+            "Service": "s3.amazonaws.com"
+        }
+    }, {
+        "Action": "sts:AssumeRole",
+        "Effect": "Allow",
+        "Principal": {
+            "Service": "logs.amazonaws.com"
+        },
+        "Sid": ""
+    }])
+    attach_policy_arns.extend([
+        "arn:aws:iam::aws:policy/CloudWatchFullAccess",
+        "arn:aws:iam::aws:policy/AmazonSSMFullAccess"
```

Comment:
We should be able to replace these with some recently-introduced, less-permissive policies: CloudWatchAgentServerPolicy, CloudWatchAgentAdminPolicy, and AmazonSSMManagedInstanceCore (see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/create-iam-roles-for-cloudwatch-agent.html).